tests/end_to_end_test/snapshot_fail_test.go

```diff
@@ -36,22 +37,27 @@ func TestSnapshotNonexistent(t *testing.T) {
 
 func TestSnapshotFail_Default(t *testing.T) {
 	t.Parallel()
-	testSnapshotFail(t, false, nil, nil)
+	testSnapshotFailText(t, false, nil, nil)
+}
+
+func TestSnapshotFail_DefaultJSONOutput(t *testing.T) {
```

Comment:
This is the new test for the change.

Everything else is refactoring to parse the JSON output in tests.